c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -1,22 +1,607 @@
+
+    /* AES is our preferred symmetric cipher */
+    auto aes = {Encryption::AES128, Encryption::AES128GCM, Encryption::AES256,
+                Encryption::AES256GCM};
+
+    /* Now arrange all ciphers by preference: */
+    it = std::stable_partition(it, ciphers.end(), byEncryption(aes));
+
+    /* Move ciphers without forward secrecy to the end */;
+    std::stable_partition(it, ciphers.end(), [compare = byKeyExchange(KeyExchange::RSA)](auto &c) {
+      return !compare(c);
+    });
+  }
+
+  std::function<bool(const Cipher &)> byProtocol(Protocol val) {
+    return [val](auto &c) { return c.protocol == val; };
+  }
+
+  std::function<bool(const Cipher &)> byKeyExchange(KeyExchange val) {
+    return [val](auto &c) { return c.kx == val; };
+  }
+
+  std::function<bool(const Cipher &)> byAuthentication(Authentication val) {
+    return [val](auto &c) { return c.au == val; };
+  }
+
+  std::function<bool(const Cipher &)> byEncryption(std::set<Encryption> vals) {
+    return [vals](auto &c) { return vals.find(c.enc) != vals.end(); };
+  }
+
+  std::function<bool(const Cipher &)> byEncryption(Encryption val) {
+    return [val](auto &c) { return c.enc == val; };
+  }
+
+  std::function<bool(const Cipher &)> byEncryptionLevel(EncryptionLevel val) {
+    return [val](auto &c) { return c.level == val; };
+  }
+
+  std::function<bool(const Cipher &)> byMessageDigest(MessageDigest val) {
+    return [val](auto &c) { return c.mac == val; };
+  }
+
+  std::vector<std::string_view> split(std::string_view s, std::string_view delimiter) {
+    size_t pos_start = 0, pos_end, delim_len = delimiter.length();
+    std::string token;
+    std::vector<std::string_view> res;
+
+    while ((pos_end = s.find(delimiter, pos_start)) != std::string::npos) {
+      token = s.substr(pos_start, pos_end - pos_start);
+      pos_start = pos_end + delim_len;
+      res.push_back(token);
+    }
+
+    res.push_back(s.substr(pos_start));
+    return res;
+  }
+
+  std::pair<std::string_view, std::string_view> split_on(std::string_view str, char c) {
+    auto ix = str.find(c);
+    if (ix == str.npos) {
+      return {str, ""};
+    }
+
+    auto left = str.substr(0, ix);
+    ix++;
+    if (ix >= str.size()) {
+      return {left, ""};
+    }
+
+    return {left, str.substr(ix)};
+  }
+
+  std::vector<std::string_view> parts(std::string_view string) {
+    std::vector<std::string_view> result;
+
+    while (!string.empty()) {
+      auto [line, rest] = split_on(string, ':');
+      string = rest;
+
+      while (!line.empty()) {
+        auto [part, rest] = split_on(line, ',');
+        line = rest;
+        result.push_back(part);
+      }
+    }
+
+    return result;
+  }
+
+  std::vector<std::string_view> splitCipherSuiteString(std::string_view string) {
+    std::vector<std::string_view> result;
+    for (auto part : parts(string)) {
+      result.push_back(part);
+    }
+    return result;
```

Comment:
It might be worth renaming `parts` to `splitCipherSuiteString` if it's not needed elsewhere.
```suggestion
    return parts(string);
```